tests/test_magpie_api.py

```diff
@@ -107,6 +111,155 @@ def setUpClass(cls):
         cls.check_requirements()
         cls.setup_test_values()
 
+    def test_Webhook_CreateUser(self):
+        """
+        Test creating a user using multiple webhooks.
+        """
+        # Create temporary config for testing webhooks
+        with tempfile.NamedTemporaryFile(mode="wt") as tmp_config:
+            base_webhook_url = get_constant("MAGPIE_TEST_USER_WEBHOOK_URL")
+            create_webhook_url = base_webhook_url + "/webhook"
+            data = {
+                "webhooks":
+                    # Use two identical urls to simulate having multiple webhook urls
+                    {"create": [create_webhook_url, create_webhook_url]}}
+            yaml.dump(data, tmp_config, default_flow_style=False)
+
+            with mock.patch.dict(os.environ, {"MAGPIE_CONFIG_PATH": tmp_config.name}):
+                utils.get_test_webhook_app()
+
+                utils.TestSetup.create_TestUser(self, override_group_name=get_constant("MAGPIE_ANONYMOUS_GROUP"))
+
+                # Wait for the webhook requests to complete
+                sleep(1)
+
+                # Check if both webhook requests have completed successfully
+                resp = requests.get(base_webhook_url + "/get_status")
+                assert resp.text == "2"
+
+                # Check if user creation was successful
+                users = utils.TestSetup.get_RegisteredUsersList(self)
+                utils.check_val_is_in(self.test_user_name, users, msg="Test user should exist.")
+
+    def test_Webhook_CreateUser_FailingUrl(self):
+        """
+        Test creating a user using a failing webhook url.
+        """
+        # Create temporary config for testing webhooks
+        with tempfile.NamedTemporaryFile(mode="wt") as tmp_config:
+            create_webhook_url = "failing_url"
```

Comment:
We should also prepare an endpoint in the webhook app that will call the callback. 
In this test, calling the url will failed and raise an exception, but calling successfully a webhook and being called back is another thing that should be tested. Right now, we know that the callback will not be processed and thus the check will failed, but at least everything could be in place already.

Reply:
I'm not 100% sure to understand here.
Do you mean that I should implement the use of the temp_url for the testing app, that would call the error_callback function?
And that I should add a test that would test that part : a webhook call that works, but resulting in an error in the webhook app internal process which would trigger the call of the temp_url and subsequently, the call of the error_callback function?

Reply:
Exact! I'm not sure if the mocking can allow such a usecase though.

Reply:
Yes, can be done, using `with mock.patch("pkg.function") as mocked:`. Can also give `side_effect=func` parameter if you want to substitute the function to do other stuff.